Core/scripts/dirac-create-distribution-tarball.py

```diff
@@ -120,47 +135,82 @@ def __discoverVCS( self ):
+    
+    :param str moduleName: The name of the Module: for example: LHCbWebDIRAC
+    :param str sourceURL: The code repository: ssh://git@gitlab.cern.ch:7999/lhcb-dirac/LHCbWebDIRAC.git
+    :param str tagVersion: the tag for example: v4r3p6
+    """
     if not self.params.vcs:
       if not self.__discoverVCS():
         return S_ERROR( "Could not autodiscover VCS" )
     gLogger.info( "Checking out using %s method" % self.params.vcs )
 
     if self.params.vcs == "file":
-      return self.__checkoutFromFile()
+      return self.__checkoutFromFile( moduleName, sourceURL )
     elif self.params.vcs == "cvs":
-      return self.__checkoutFromCVS()
+      return self.__checkoutFromCVS( moduleName, sourceURL )
     elif self.params.vcs == "svn":
-      return self.__checkoutFromSVN()
+      return self.__checkoutFromSVN( moduleName, sourceURL, tagVersion )
     elif self.params.vcs == "hg":
-      return self.__checkoutFromHg()
+      return self.__checkoutFromHg( moduleName, sourceURL )
     elif self.params.vcs == "git":
-      return self.__checkoutFromGit()
+      return self.__checkoutFromGit( moduleName, sourceURL, tagVersion )
 
     return S_ERROR( "OOPS. Unknown VCS %s!" % self.params.vcs )
 
-  def __checkoutFromFile( self ):
-    sourceURL = self.params.sourceURL
+  def __checkoutFromFile( self, moduleName = None, sourceURL = None ):
+    """
+    This method checkout a given tag from a file 
+    Note: we can checkout any project form a file 
+    
+    :param str moduleName: The name of the Module
+    :param str sourceURL: The code repository
+        
+    """
+    if not moduleName:
+      moduleName = self.params.name
+    
+    if not sourceURL:
+      sourceURL = self.params.sourceURL   
+      
     if sourceURL.find( "file://" ) == 0:
       sourceURL = sourceURL[ 7: ]
     sourceURL = os.path.realpath( sourceURL )
     try:
       pyVer = sys.version_info
       if pyVer[0] == 2 and pyVer[1] < 6:
         shutil.copytree( sourceURL,
-                         os.path.join( self.params.destination, self.params.name ),
+                         os.path.join( self.params.destination, moduleName ),
                          symlinks = True )
       else:
         shutil.copytree( sourceURL,
-                         os.path.join( self.params.destination, self.params.name ),
+                         os.path.join( self.params.destination, moduleName ),
                          symlinks = True,
                          ignore = shutil.ignore_patterns( '.svn', '.git', '.hg', '*.pyc', '*.pyo', 'CVS' ) )
     except Exception as e:
       return S_ERROR( "Could not copy data from source URL: %s" % str( e ) )
     return S_OK()
 
-  def __checkoutFromCVS( self ):
-    cmd = "cvs export -d '%s' '%s'" % ( self.params.sourceURL, os.path.join( self.params.destination, self.params.name ) )
+  def __checkoutFromCVS( self, moduleName = None, sourceURL = None ):
```

Comment:
Shall we go on and drop this? (I understand this is out of the scope of this PR, but still... CVS?)

Reply:
We have to decide, which version controls systems are supported by DIRAC and we have to clean the code. I want to make everything consistent, that's why I modified this method.
We are going to support:git, mercurial, svn? Maybe we can drop svn as well? I can make a cleanup, if we decide the supported version control systems.

Reply:
I think we can safely drop CVS everywhere. At the same time I am not sure we can fully drop SVN.

Reply:
Okay. I clean all CVS related code. We will decide what to do with svn